crates/l1tx/src/filter/withdrawal_fulfillment.rs

```diff
@@ -0,0 +1,254 @@
+use bitcoin::{ScriptBuf, Transaction};
+use strata_primitives::l1::{BitcoinAmount, WithdrawalFulfillmentInfo};
+
+use super::TxFilterConfig;
+use crate::utils::op_return_nonce;
+
+fn create_opreturn_metadata(operator_idx: u32, deposit_idx: u32) -> ScriptBuf {
+    let mut metadata = [0u8; 8];
+    // first 4 bytes = operator idx
+    metadata[..4].copy_from_slice(&operator_idx.to_be_bytes());
+    // next 4 bytes = deposit idx
+    metadata[4..].copy_from_slice(&deposit_idx.to_be_bytes());
+    op_return_nonce(&metadata)
+}
+
+/// Parse transaction and search for a Withdrawal Fulfillment transaction to an expected address.
+pub fn parse_withdrawal_fulfillment_transactions<'a>(
+    tx: &'a Transaction,
+    filter_conf: &'a TxFilterConfig,
+) -> Option<WithdrawalFulfillmentInfo> {
+    // 1. Check this is of correct structure
+    let frontpayment_txout = tx.output.first()?;
+    let metadata_txout = tx.output.get(1)?;
+    if !metadata_txout.script_pubkey.is_op_return() {
+        return None;
+    }
+
+    // 2. Check withdrawal is to an address we expect
+    let withdrawal = filter_conf
+        .expected_withdrawal_fulfillments
+        .binary_search_by_key(&frontpayment_txout.script_pubkey, |expected| {
+            expected.destination.inner()
+        })?;
+
+    // 3. Ensure amount is equal to the expected amount
+    let actual_amount_sats = frontpayment_txout.value.to_sat();
+    if actual_amount_sats < withdrawal.amount {
```

Comment:
This is actually a somewhat thought-provoking condition.  What if it's *more* than the requested amount?

Reply:
Can't we just simply consider it fulfilled if an operator pays more than needed?